src/utils/tokenUtils.ts

```diff
@@ -296,14 +296,45 @@ export const getTokenBalance = async (userId: string, tokenData: ITokenData) =>
 };
 
 export const getTokenPrice = (poolsData: IPoolData[], tokenAddress: string, hbarPrice: number) => {
+  const WHBARAddress = process.env.REACT_APP_WHBAR_ADDRESS;
+  const minLiquidity = Number(process.env.REACT_APP_POOL_MIN_LIQUIDITY as string);
+
   if (hbarPrice === 0) return '0';
-  if (tokenAddress === process.env.REACT_APP_WHBAR_ADDRESS) return hbarPrice.toString();
+  if (tokenAddress === WHBARAddress) return hbarPrice.toString();
+
+  // Filter pools with WHBAR not deep enough in order to not distort price calculation
+  const filteredPoolsData = poolsData.filter(pool => {
+    const { token0, token0Amount, token0Decimals, token1, token1Amount, token1Decimals } = pool;
+
+    const token0AmountFormattedNumber = Number(
+      formatStringWeiToStringEther(token0Amount, token0Decimals),
+    );
```

Comment:
I think you can get this from the response (token0AmountFormatted and token1AmountFormatted)

Reply:
I checked it, although it's not so trivial, cause I need to update an interface `IPoolData` to `IPoolDataExtended` and will be kind of tricky...